service_configuration_lib/spark_config.py

```diff
@@ -297,7 +299,13 @@ def _adjust_spark_requested_resources(
     elif cluster_manager == 'kubernetes':
         # TODO(gcoll|COREML-2697): Consider cleaning this part of the code up
         # once mesos is not longer around at Yelp.
-        if 'spark.executor.instances' not in user_spark_opts:
+        if (
+                'spark.executor.instances' not in user_spark_opts and
+                (
+                    'spark.dynamicAllocation.enabled' not in user_spark_opts or
```

Comment:
What happens when `spark.dynamicAllocation.enabled` is `true`?
How would the `spark.executor.instances` be calculated then?